container/tabs.go

```diff
@@ -269,6 +272,18 @@ func (r *baseTabsRenderer) minSize(t baseTabs) fyne.Size {
 }
 
 func (r *baseTabsRenderer) moveIndicator(pos fyne.Position, siz fyne.Size, animate bool) {
+	r.indicator.Show()
+	if r.indicator.Position().IsZero() {
+		r.indicator.Move(pos)
+		r.indicator.Resize(siz)
+		r.indicator.Refresh()
+		return
+	}
+
+	if r.lastIndicatorPos.Subtract(pos).IsZero() && r.lastIndicatorSize.Subtract(siz).IsZero() {
+		return
+	}
+
```

Comment:
Are you sure these early-exit conditions should be before stopping any previous animations?

Reply:
I think, it is needed, otherwise it will start animations as before even if there are others running with the same end state 🤔

Reply:
Oh yes I agree they are needed, I'm just wondering if they should come after these;
```go
	if r.positionAnimation != nil {
		r.positionAnimation.Stop()
		r.positionAnimation = nil
	}
	if r.sizeAnimation != nil {
		r.sizeAnimation.Stop()
		r.sizeAnimation = nil
	}
```
So that even if we exit early, we still stop any existing animations

Reply:
The thing is that if we put that after that conditions, the current animation will stop and it is not going to reach its end state

Reply:
Do we want the current animations to reach their end state if we have explicitly moved & resized the indicator?
```go
		r.indicator.Move(pos)
		r.indicator.Resize(siz)
```

Reply:
The original code was;
```go
	if r.positionAnimation != nil {
		r.positionAnimation.Stop()
		r.positionAnimation = nil
	}
	if r.sizeAnimation != nil {
		r.sizeAnimation.Stop()
		r.sizeAnimation = nil
	}

	r.indicator.Show()
	if r.indicator.Position().IsZero() {
		r.indicator.Move(pos)
		r.indicator.Resize(siz)
		r.indicator.Refresh()
		return
	}
```
I don't mind it getting changed, but I want to make sure there isn't a possibility that the indicator is correctly moved & resized, but a previous animation is still running and overwrites the correct values with wrong values.

Reply:
Sorry, I don't understand. You mean this part:
```go
r.indicator.Show()
	if r.indicator.Position().IsZero() {
		r.indicator.Move(pos)
		r.indicator.Resize(siz)
		r.indicator.Refresh()
		return
	}
```
should go in a different place?

Reply:
> a possibility that the indicator is correctly moved & resized, but a previous animation is still running and overwrites the correct values with wrong values.

Hmm, Indicator will be moved and resized correctly, no previous animation can overwrites them with wrong values because it will only happens if lastIndicatorPos and Siz have changed, and if they have changed, then we stop animations if they are not nil 🤔

Reply:
However, I wonder if a mutex is needed to guarantee that lastValues are not written simultaneously, what do you think?

Reply:
I mean like this;
```go
	if r.lastIndicatorPos.Subtract(pos).IsZero() && r.lastIndicatorSize.Subtract(siz).IsZero() {
		return
	}

	if r.positionAnimation != nil {
		r.positionAnimation.Stop()
		r.positionAnimation = nil
	}
	if r.sizeAnimation != nil {
		r.sizeAnimation.Stop()
		r.sizeAnimation = nil
	}

	r.indicator.Show()
	if r.indicator.Position().IsZero() {
		r.indicator.Move(pos)
		r.indicator.Resize(siz)
		r.indicator.Refresh()
		return
	}
```

Reply:
Yes, you are write that should be below, I will update it :)

Reply:
Cool!

> I wonder if a mutex is needed to guarantee that lastValues are not written simultaneously

Yes, I think you maybe right.

Reply:
Ok, I will add them and then test :)

Reply:
Done.